### PR TITLE
feat: Add a general `text_wrapper` to create `repr` methods

### LINKED ISSUE #2259: [FEATURE] add `__repr__` for the SDK classes
**Is your feature request related to a problem? Please describe.**
N.A.

**Describe the solution you'd like**
The method should expose the active workspace and potentially available workspaces, URL/host and available datasets.
Suggestions are welcome:)

**Describe alternatives you've considered**
N.A.

**Additional context**
N.A.

### PR DESCRIPTION
**Pull request by kursathalat**
# Description

This PR adds a `text_wrapper` method that creates object representations automatically for classes that inherits from pydantic `BaseModel`. It also includes changes in repr methods of some other classes that are not inheriting from BaseModel.

Closes #2259 

**Type of change**

(Please delete options that are not relevant. Remember to title the PR according to the type of change)

- [ ] New feature (non-breaking change which adds functionality)
- [ ] Refactor (change restructuring the codebase without changing functionality)
- [ ] Improvement (change adding some improvement to an existing functionality)

**How Has This Been Tested**

(Please describe the tests that you ran to verify your changes. And ideally, reference `tests`)

- [ ] Test A
- [ ] Test B

**Checklist**

- [ ] I added relevant documentation
- [ ] I followed the style guidelines of this project
- [ ] I did a self-review of my code
- [ ] I made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I filled out [the contributor form](https://tally.so/r/n9XrxK) (see text above)
- [ ] I have added relevant notes to the `CHANGELOG.md` file (See https://keepachangelog.com/)

### CODE REVIEW

File: src/argilla/textwrapping.py [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+    Returns:
+        A string for the __repr__ method of the class that inherits from pydantic.BaseModel.
+    """
+
+    if attr_ignore == None:
+        attr_ignore = []
+    if attr_multiline == None:
+        attr_multiline = []
+
+    def create_wrapped_list(self, attr_ignore=None, attr_multiline=None):
```
Comment (davidberenstein1957) [inline src/argilla/textwrapping.py:36]: would it be possible to check if an attribute is a `str` and then limit the entire printed text? To the first n characters ... last n characters?

File: src/argilla/listeners/models.py [hunk 2]
```diff
@@ -36,6 +37,15 @@ class Search:
     total: int
     query_params: Optional[Dict[str, Any]] = None
 
+    def __repr__(self) -> str:
```
Comment (davidberenstein1957) [inline src/argilla/listeners/models.py:40]: we added a textwrapper to avoid this right?

File: src/argilla/listeners/models.py [hunk 3]
```diff
@@ -67,6 +77,16 @@ class RGListenerContext:
     metrics: Optional[Metrics] = None
     query_params: Optional[Dict[str, Any]] = None
 
+    def __repr__(self) -> str:
```
Comment (davidberenstein1957) [inline src/argilla/listeners/models.py:80]: we added a textwrapper to avoid this right?

File: src/argilla/client/workspaces.py [hunk 4]
```diff
@@ -119,9 +120,14 @@ def users(self) -> List["UserModel"]:
         return workspaces_api.list_workspace_users(self._client, self.id).parsed
 
     def __repr__(self) -> str:
```
Comment (davidberenstein1957) [inline src/argilla/client/workspaces.py:122]: we added a textwrapper to avoid this right?

File: src/argilla/client/users.py [hunk 5]
```diff
@@ -149,11 +150,19 @@ def is_annotator(self) -> bool:
         return self.role == UserRole.annotator
 
     def __repr__(self) -> str:
```
Comment (davidberenstein1957) [inline src/argilla/client/users.py:152]: we added a textwrapper to avoid this right?

**Review by davidberenstein1957 — COMMENTED** [review]
LFTM but some minor remarks.

@kursathalat, as discussed we might make a share `Basemodel` across all schemas which we can overwrite. I think we can add this shared `TextWrappedBaseModel` and the `textwrapping` to `src/argilla/utils/utils.py`. 

Additionally, can you add some examples to the different representations using the arguments in the functiosns?

File: src/argilla/textwrapping.py (hunk 1 above)
Comment (kursathalat) [reply src/argilla/textwrapping.py:36]: Currently, it is limiting `str` longer than 50, we can change it. However some strings like name are `ConstrainedStrValue`, so they are not limited and they should be also be addressed.

**Comment by kursathalat** [comment]
Currently it is 
```
repr(rg.SuggestionSchema(question_name="quest")) # type is not shown
repr(rg.VectorSettings(name="vectorsetting", title="An example of a string that is longer than 50 characters so this one is limited as the first and second part which are connected with three dots.", dimensions=5))
```
and output

```
SuggestionSchema(
   question_name=quest
   score=None
   value=None
   agent=None
)
VectorSettings(
   name=vectorsetting
   title=An example of a string ... ected with three dots.
   dimensions=5
)
```

The problem is that the process of ignoring attributes is done via the `default` parameter of `Field` at the moment, which is temporary and should be changed. Ideally, we want to access the `repr` parameter of `Field`, but have not been able to do it.

**Pull request closed without merging**